src/components/Input/Input.tsx

```diff
@@ -1,73 +1,179 @@
-import React from 'react';
+import * as React from 'react';
+import { cva, type VariantProps } from 'class-variance-authority';
 import { cn } from '../../helpers/classnames';
 import { Tooltip } from 'react-tooltip';
 
-interface InputProps {
+const inputVariants = cva(
+  [
+    'block',
+    'w-full',
+    'rounded-md',
+    'border-0',
+    'px-2',
+    'py-1.5',
+    'focus:ring-0',
+    'disabled:bg-background-300/50',
+    'disabled:text-text-700/50',
+    'disabled:ring-primary-300',
+  ],
+  {
+    variants: {
+      variant: {
+        outline: [
+          'border-secondary-500',
+          'bg-background-50',
+          'ring-1',
+          'ring-inset',
+          'ring-secondary-300',
+          'placeholder:text-text-700',
+          'hover:ring-primary-600',
+        ],
+        error: [
+          'text-red-700',
+          'ring-1',
+          'ring-red-600',
+          'hover:ring-red-800',
+          'focus:outline-red-600',
+        ],
+      },
+      adornment: {
+        start: ['pl-10'],
+        end: ['pr-10'],
+      },
+    },
+    defaultVariants: {
+      variant: 'outline',
+    },
+  },
+);
+
+export interface BaseInputProps
+  extends React.InputHTMLAttributes<HTMLInputElement>,
+    VariantProps<typeof inputVariants> {
   id: string;
```

Comment:
Should this be a required parameter? I feel like it should be optional, unless a label is provided

Reply:
It is now optional and if the label is set then Id is required.